File: 30daysOFcode_Swift/day 18.swift
struct Stack<Element> {
    var items = [Element]()
    mutating func push(_ item: Element) {
        items.append(item);
    }
    mutating func pop() -> Element {
        return items.removeLast();
    }
}

struct Queue<Element> {
    var items = [Element]()
    mutating func enqueue(_ item: Element) {
        items.append(item);
    }
    mutating func dequeue() -> Element {
        return items.removeFirst();
    }
}


class Solution {

    var stack = Stack<Character>();
    var queue = Queue<Character>();

    func pushCharacter(ch: Character){
        stack.push(ch);
    }
    
    func popCharacter() -> Character {
        return stack.pop();
    }
    
    func enqueueCharacter(ch: Character){
        queue.enqueue(ch);
    }
    
    func dequeueCharacter() -> Character {
        return queue.dequeue();
    }
}

File: 30daysOFcode_TypeScript/day 18.ts
class Solution {
    stack: Array<string>;
    queue: Array<string>;

    constructor() {
        this.stack = [];
        this.queue = [];
    }

    pushCharacter(c: string): void {
        this.stack.push(c);
    }

    popCharacter(): string {
        return this.stack.pop();
    }

    enqueueCharacter(c: string): void {
        this.queue.push(c);
    }

    dequeueCharacter(): string {
        return this.queue.shift();
    }
}
